sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/LoadBalancerOutboundRuleImpl.java

```diff
@@ -46,19 +48,17 @@ public LoadBalancerOutboundRuleProtocol protocol() {
     }
 
     @Override
-    public List<LoadBalancerFrontend> frontends() {
-        List<LoadBalancerFrontend> retFrontEnds = new ArrayList<>();
+    public Map<String, LoadBalancerFrontend> frontends() {
+        Map<String, LoadBalancerFrontend> nameToFrontEndMap = new TreeMap<>();
         if(this.innerModel().frontendIpConfigurations() != null && !this.innerModel().frontendIpConfigurations().isEmpty()) {
             for(SubResource frontendIpConfiguration : this.innerModel().frontendIpConfigurations()) {
-                retFrontEnds.add(
-                    this
-                    .parent()
+                LoadBalancerFrontend frontend = this.parent()
                     .frontends()
-                    .get(ResourceUtils.nameFromResourceId(frontendIpConfiguration.id()))
-                );
+                    .get(ResourceUtils.nameFromResourceId(frontendIpConfiguration.id()));
+                nameToFrontEndMap.put(frontend.name(), frontend);
             }
         }
-        return retFrontEnds;
+        return nameToFrontEndMap;
```

Comment:
Do you need immutable here as well?

Reply:
may be better to make it immutable, will add